src/H5Dmpio.c

```diff
@@ -1376,124 +1666,126 @@ H5D__link_chunk_filtered_collective_io(H5D_io_info_t *io_info, const H5D_type_in
      */
     H5CX_set_mpio_actual_io_mode(H5D_MPIO_CHUNK_COLLECTIVE);
 
+    /* Enable collective metadata reads if not already enabled. The collective
+     * re-insertion of modified chunks into the chunk index does not scale well
+     * without reading metadata collectively and it should be safe to enable
+     * them here because it is required that these dataset writes be collective.
+     */
+    H5CX_set_coll_metadata_read(TRUE);
```

Comment:
The changes here currently turn on collective metadata reads for the feature since it will not scale well without them and we require the writes to be collective, but this may or may not be the best idea depending on the number of chunks in the operation.